src/stdatamodels/jwst/transforms/models.py

```diff
@@ -1989,6 +1991,161 @@ def _evaluate_transform_guess_form(model, x=0, y=0, t=0):
     raise TypeError(f"Expected a model or list of models, but got {type(model)}. ")
 
 
+class MIRIWFSSBackwardDispersion(_BackwardGrismDispersionBase):
+    """Calculate the dispersion extent of MIRI WFSS pixels."""
+
+    def __init__(self, orders, lmodels=None, xmodels=None, ymodels=None, name=None):
+        """
+        Initialize the model.
+
+        Parameters
+        ----------
+        orders : list
+            The list of orders which are available to the model. For MIRIWFSS orders = 1.
+        lmodels : list
+            The list inverse dispersion 1D polynomial models
+        xmodels : list
+            The list of 1D polynomial model in x.
+        ymodels : list[tuple]
+            The list of 2D polynomial models in y.
+            There are 3 2D polynomials to correct for  spatial-dependence.
```

Comment:
```suggestion
        ymodels : list[list]
            The list of 2D polynomial models in y. The length of ymodels must be
            equal to the length of orders. Each element in ymodels is a sublist of
            3 2D polynomials to correct for spatial-dependence.
```
Am I understanding the code here that?
- orders should be a list of integers with length 1
- lmodels and xmodels are lists of 1D polynomials with lengths equal to the length of orders (N=1)
- ymodels is a list with length equal to the length of orders (N=1) and containing elements that are 3 item long lists containing 2D polynomials

Also, I'm confused by `correct for spatial-dependence`. Correct for spatial-dependence of what? Are there some more details you can add here to help me understand what these are used for?

Reply:
I can help here.  This all looks correct to me.

Orders being a list makes sense for the other instruments where you can have multiple spectral orders, each of which has their own trace function.  I think it's better for MIRI to keep orders as a list for consistency in call structure between the instruments (but I could be convinced otherwise)

lmodels, xmodels, and ymodels are all functions of the trace parameter `t`, such that `lmodels` defines wavelength(t), `xmodels` defines `x(t)`, and `ymodels` defines `y(t)`.  These functions can have arbitrary complexity, where more complex would be a better representation of the true spectral trace.  It turns out that the three functions need to have different amounts of complexity in order to approximate the "true" trace.

Specifically, the `lmodel` and `xmodel` can be approximated by a simple 1st order polynomial in `t` for MIRI, but the `ymodel` is more complicated.  The `ymodel` depends not only on `t` but also on the `x0, y0` pixel location in the direct image that you're dispersing into the grism frame.  So each of the coefficients `C0`, `C1`, and `C2` of the second-order polynomial in `t` themselves have dependence on the direct-image position, so the full functional form looks something like `y_dispersed = C0(x0, y0) + C1(x0, y0)*t + C2(x0, y0)*t**2`

Reply:
@emolter Thank you for explaining that. It is a bit complex. Let me see if I can add more details